### ISSUE TITLE
cobbler-completion missing

### ISSUE THREAD

**Issue by jehutyy** [cobbler/cobbler#2042]
It seems that the file **/etc/cobbler/cobbler_bash** provided uses cobbler-completion program but this one doesn't exist on the system. 

Even **yum provides  cobbler-completion** or **yum search cobbler-completion** returns nothing.

**Comment by brejoc** [comment]
At least the [file is still there](https://github.com/cobbler/cobbler/blob/master/config/bash/cobbler_bash). Could you please check your RPM if it is included? Which distro are we talking about here?

**Comment by jehutyy** [comment]
I'm using centos 7 and fedora 29. I installed cobbler from default yum repo.
As you said cobbler_bash is here but as you can see line 33 : 

SPECIAL_CMDS=`cobbler-completion | sed "s/ /|/g"`

But the cobbler-completion command is not provided. Thus sourcing cobbler_bash and trying to complete cobbler command result in "cobbler-completion not found".

**Comment by xenlo** [comment]
Facing the same issue in OpenSUSE Leap 15.1!

### OS Version ###
```
cobbler-server:~ # cat /etc/os-release
NAME="openSUSE Leap"
VERSION="15.1 "
ID="opensuse-leap"
ID_LIKE="suse opensuse"
VERSION_ID="15.1"
PRETTY_NAME="openSUSE Leap 15.1"
[…]
```

### Reproduce ###
```
cobbler-server:~ #  zypper install cobbler
[…]
cobbler-server:~ # cp /etc/cobbler/cobbler_bash /etc/bash_completion.d/cobbler
cobbler-server:~ # bash
cobbler-server:~ # cobbler sys<tab>cobbler-completion: command not found
cobbler-server:~ # rpm -ql /var/cache/zypp/packages/repo-oss/noarch/cobbler-2.8.2-lp151.4.2.noarch.rpm | grep cobbler-completion
cobbler-server:~ # echo $?
1
```

**Comment by xenlo** [comment]
Actually I just found that actually the `cobbler-completion` has been deleted from the repo (because not finished?).

Commit: https://github.com/cobbler/cobbler/commit/a70ee7ad33d588fa8b901d7108869c02b1f384ed

**Comment by SchoolGuy** [comment]
Hm I would say this is a part which would improve the experience with cobbler. So let's bring it back and finish it/recreate it!

**Comment by jehutyy** [comment]
I made an attempt on writing a cobbler-completion.bash few months ago which kind of work despite being "well-coded". Tell me if you're interested in reviewing and integrating it.

**Comment by SchoolGuy** [comment]
@jehutyy Of course! Start a PR and we will have a look. Even if you cannot complete it we will do it for you!

**Comment by jehutyy** [comment]
I just made a PR. I don't know how to add reviewers so don't hesitate to have a look and get back to me for improvement ;)

**Comment by xenlo** [comment]
I don't know when, but I will definitively give it a try to your PR #2208.

**Comment by xenlo** [comment]
As promised, I tested your completion script. Thanks for removing me a so old frustration. 
👍 
I added several comments. Most of them are improvement. But that will not take you too much time as I provided a suggestion each time that I tested in my environment already (which is a cobbler 2.8.2). 

Here is the gists of the [file with all my suggestions](https://gist.github.com/xenlo/2f68ce6bf03ce7eb96f3154ade1ba713) if it could help you.
